runtime/caml/misc.h

```diff
@@ -98,6 +98,8 @@ CAMLdeprecated_typedef(addr, char *);
 #define CAMLalign(n) _Alignas(n)
 #elif defined(SUPPORTS_ALIGNED_ATTRIBUTE)
 #define CAMLalign(n) __attribute__((aligned(n)))
+#elif defined(__cplusplus) && (_MSC_VER >= 1900 || __cplusplus > 199711L)
```

Comment:
`alignas` is C++ 2011, if I'm not mistaken, so the test should be `__cplusplus > 201103L`.

Reply:
Also I would put this case above the `#elif defined(SUPPORTS_ALIGNED_ATTRIBUTE)` so that the C++-standard `alignas` implementation is preferred over the compiler-specific `__attribute__` implementation.

Reply:
> `alignas` is C++ 2011, if I'm not mistaken, so the test should be `__cplusplus >= 201103L`.

You’re right, but there’s no new value between 199711L and 201103L. I’ll change the value, but the result is the same.

> Also I would put this case above the `#elif defined(SUPPORTS_ALIGNED_ATTRIBUTE)` so that the C++-standard `alignas` implementation is preferred over the compiler-specific `__attribute__` implementation.

Sure!

Reply:
[Stroustroup suggests `> 199711L` for detecting C++11](https://www.stroustrup.com/C++11FAQ.html#11) I have no idea whether there are actually an C++ compilers for which that would actually make a difference (just for the context of where the test was taken).

Reply:
Stroustrup's text reads like it was written before the exact value of `__cplusplus` for C11 was determined.

Not being a C++ expert, I cannot guess that `__cplusplus > 199711L` means C++11 or later, while I can guess that `__cplusplus >= 201103L` means this.